src/Motions/MatchPair.ts

```diff
@@ -0,0 +1,57 @@
+import {window, Position, Range} from 'vscode';
+import {Motion} from './Motion';
+import {TextObjectBlock} from '../TextObjects/Block';
+
+enum MotionMatchDirection {NEXT, PREV};
+
+export class MotionMatchPair extends Motion {
+    private static openingCharacterMap = {
+        '(': TextObjectBlock.byParentheses,
+        '{': TextObjectBlock.byBraces,
+        '[': TextObjectBlock.byBrackets,
+    }
+    
+    private static closingCharacterMap = {
+        ')': TextObjectBlock.byParentheses,
+        '}': TextObjectBlock.byBraces,
+        ']': TextObjectBlock.byBrackets,
+    }
+    
+    private static characterMap = Object.assign(
+        {}, MotionMatchPair.openingCharacterMap, MotionMatchPair.closingCharacterMap
+    );
+
+    static matchPair(): Motion {
+        const obj = new MotionMatchPair();
+        return obj;
+    }
+
+    apply(from: Position, option: {isInclusive?: boolean} = {}): Position {
+        option.isInclusive = option.isInclusive === undefined ? false : option.isInclusive;
+
+        from = super.apply(from);
+
+        const activeTextEditor = window.activeTextEditor;
+
+        if (!activeTextEditor) {
+            return from;
+        }
+
+        const document = activeTextEditor.document;
+        
+        const characterUnderCursor = document.getText(new Range(from, new Position(from.line, from.character+1)));
```

Comment:
Sure.
I think it would be more efficient to do it like this:

```ts
document.getText(new Range(
    anchor.line, anchor.character,
    anchor.line, anchor.character + 1
));
```